src/session.cpp

```diff
@@ -325,29 +325,27 @@ namespace libtorrent
 
 	namespace
 	{
-		std::vector<boost::shared_ptr<plugin>> default_plugins()
+		std::vector<boost::shared_ptr<plugin>> default_plugins(
+			bool empty = false)
 		{
-			std::vector<boost::shared_ptr<plugin>> v;
 #ifndef TORRENT_DISABLE_EXTENSIONS
+			if (empty) return {};
 			using wrapper = session_impl::session_plugin_wrapper;
-			boost::shared_ptr<plugin> ut_pex(new wrapper(create_ut_pex_plugin));
-			boost::shared_ptr<plugin> ut_metadata(new wrapper(create_ut_metadata_plugin));
-			boost::shared_ptr<plugin> smart_ban(new wrapper(create_smart_ban_plugin));
-
-			v.push_back(ut_pex);
-			v.push_back(ut_metadata);
-			v.push_back(smart_ban);
+			return {
+				boost::make_shared<wrapper>(wrapper(create_ut_pex_plugin)),
+				boost::make_shared<wrapper>(wrapper(create_ut_metadata_plugin)),
+				boost::make_shared<wrapper>(wrapper(create_smart_ban_plugin))
+			};
+#else
+			return {};
 #endif
-			return v;
 		}
 	}
 
-	void session::start(int const flags, settings_pack const sp, io_service* ios)
+	void session::start(int flags, settings_pack sp, io_service* ios)
```

Comment:
I do think it's reasonable for the flags to be const here though.. but I don't think sp can be const, since we're moving it (in fact, I'm a bit surprised std::move() accepts a const value)